plugins/com.google.cloud.tools.eclipse.projectselector/src/com/google/cloud/tools/eclipse/projectselector/model/GcpProject.java

```diff
@@ -70,19 +70,19 @@ public boolean equals(Object obj) {
    * @return an AppEngine object if it has been retrieved from the backend or {@code null} if it has
    * not yet been retrieved
    */
-  public AppEngine getAppEngine() {
+  public synchronized AppEngine getAppEngine() {
```

Comment:
assuming only a single thread sets the value of the appEngine field, we should just mark it `volatile` instead of using `synchronized`. See:

https://stackoverflow.com/questions/9749746/what-is-the-difference-between-atomic-volatile-synchronized/37395582#37395582

If multiple threads may write this, then we should use `AtomicReference` instead.